docs/contributor/adr/003-client-scope-and-usage.md

```diff
@@ -0,0 +1,37 @@
+# ADR 003 - Client scope and usage
+
+## Status
+
+Accepted
+
+## Context
+
+Our current application architecture includes direct references to the Kubernetes client interface across multiple layers, such as the Reconciler and to be Service layers. This leads to a violation of the separation of concerns and tightly couples orchestration and business logic to infrastructure-specific code.
+
+To address this, we will adopt a 3-tier architecture, where infrastructure dependencies like the Kubernetes client should only be referenced within the Repository layer. This change aligns with our architectural goal of modularity and improves testability and maintainability, ensuring a clear boundary between data access and business logic or orchestration components.
```

Comment:
Completely valid, I would prefer to have this a separate small ADR about the target architecture. This ADR is then only about decision and consequences of using `client.Client` in the repository layer.

Reply:
The target architecture is defined [here](https://github.com/kyma-project/lifecycle-manager/pull/2453), as discussed.
Still for context and layer location this is an important information. I shortened it a bit. Is it ok for you like this?